src/Shared/Recruit.Vacancies.Client/Infrastructure/Reports/ProviderApplicationsReportStrategy.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Mongo;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Polly;
+using JsonConvert = Newtonsoft.Json.JsonConvert;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Reports
+{
+    internal class ProviderApplicationsReportStrategy : MongoDbCollectionBase, IReportStrategy
+    {
+        private const string QueryUkprn = "_ukprn_";
+        private const string QueryFromDate = "_fromDate_";
+        private const string QueryToDate = "_toDate_";
+
+        private const string ColumnProgramme = "Programme";
+        private const string ColumnProgrammeStatus = "Programme_Status";
+        private const string ColumnApplicationLastUpdatedDate = "Application_LastUpdatedDate";
+        private const string ColumnApplicationDate = "Application_Date";
+        private const string ColumnNumberOfDaysAppAtThisStatus = "Number_Of_Days_App_At_This_Status";
+        private const string ColumnVacancyReference = "Vacancy_Reference_Number";
+        private const string ColumnFramework = "Framework";
+        private const string ColumnFrameworkStatus = "Framework_Status";
+        private const string ColumnStandard = "Standard";
+        private const string ColumnStandardStatus = "Standard_Status";
+
+        private readonly IApprenticeshipProgrammeProvider _programmeProvider;
+        private readonly ITimeProvider _timeProvider;
+        private readonly ILogger<ProviderApplicationsReportStrategy> _logger;
+
+        private const string QueryFormat = @"[
+            { $match: {'ownerType' : 'Provider', 'trainingProvider.ukprn' : _ukprn_}},
+            { $lookup: { from: 'applicationReviews', localField: 'vacancyReference', foreignField: 'vacancyReference', as: 'ar'}},
+            { $unwind: '$ar'},
+            { $match: {'ar.submittedDate' : { $gte: ISODate('_fromDate_'), $lt: ISODate('_toDate_')}}},
+            { $project: {
+                    '_id' : 0,
+                    'Candidate_Name' : { $concat: ['$ar.application.firstName', ' ', '$ar.application.lastName']},
+                    'Applicant_id' : '$ar._id',
```

Comment:
Not sure if the application review id, is the same as applicant id. Did you mean candidate id?

Reply:
👍 Good shout. The v1 example looks like this value is per candidate. Will change